feat(password): initial guidance (#38)
vmware-clarity/clarity-guidance: 1019-password-design-guidance.md

---
category: component design guidance
state: draft
created: 2024-10-22
updated: 2024-10-22
---

# Password Component

> A [password](https://clarity.design/documentation/password) component is a specialized input field with the ability to toggle between a masked field or to view the password in plain text.

Reviewer: ```suggestion
A [password](https://clarity.design/documentation/password) component is an input field that allows users to toggle between hiding or showing the password in plain text.
```

## Guidance

> - **Must** use a password field when the user needs to set or input the password.

Reviewer: ```suggestion
- Design teams **must** use a password field when the user needs to set or input a password. This ensures security and consistency across the interface. 
```

> - **Must not** use an asterisk (`*`) in the password field.

Reviewer: ```suggestion
- Design teams **must not** use an asterisk (*) in the password field. Instead, use dots or bullets to mask the password for consistency and clarity.
```

> - **Should** show password requirements in the helper text to guide the user.

Reviewer: ```suggestion
- Design teams **should** show password requirements in the helper text to guide the user, helping them create valid passwords without confusion.
```

> - **Should not** hide the helper requirements in a signpost or reveal them only after the user fails the first attempt.

Reviewer: ```suggestion
- Design teams **should** not hide password requirements in a signpost or reveal them only after the user fails the first attempt. Providing them upfront helps prevent frustration and improves the user experience.
```

> - **May** use a password field when the user needs to mask the input content.

Reviewer: ```suggestion
- Design teams may use a password field when the user needs to mask the input content to protect sensitive information, even if it is not a password.
```

Reviewer 2: (edited Jay's comment to bold "may")
